git-server/src/lib.rs

```diff
@@ -182,11 +229,30 @@ async fn git_handler(
     headers: HeaderMap,
     body: impl Buf,
     remote: Option<net::SocketAddr>,
-    namespace: String,
+    mut namespace: String,
     request: warp::filters::path::Tail,
     query: String,
 ) -> Result<Box<dyn Reply>, Rejection> {
     let remote = remote.expect("There is always a remote for HTTP connections");
+    if namespace[namespace.len() - 4..].eq(".git") {
```

Comment:
Perhaps you can use `String::ends_with` here

Reply:
Fixed https://github.com/radicle-dev/radicle-client-services/pull/74/commits/67c4936ad316ef7490d2c15d4b88ac055b09ffe9#diff-b672a15ad14f54528cebd77aee76d3007962d44bda987b3e0a34597ec80fd6b1R228